File: internal/asword/asword.go
package asword

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

type Sword int

const (
	Green Sword = iota
	Red
	Purple
	Black
	White
	RealRed
	Blue
)

type AncientSword struct {
	swordnow string
	swordall string
}

func NewAncientSword(now, all string) *AncientSword {
	return &AncientSword{
		swordnow: now,
		swordall: all,
	}
}

func (as *AncientSword) GetNow() Sword {
	return as.ReadNow()
}

func (as *AncientSword) GetAll() []Sword {
	return as.ReadAll()
}

func (as *AncientSword) ReadNow() Sword {
	file, err := os.OpenFile(as.swordnow, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return Green
	}
	file.Close()

	file, err = os.Open(as.swordnow)
	if err != nil {
		return Green
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	var a string
	if scanner.Scan() {
		a = scanner.Text()
	}

	tmp, err := strconv.Atoi(a)
	if err != nil || tmp < 0 || tmp > 6 {
		return Green
	}
	return Sword(tmp)
}

func (as *AncientSword) ReadAll() []Sword {
	file, err := os.OpenFile(as.swordall, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return []Sword{Green}
	}
	file.Close()

	file, err = os.Open(as.swordall)
	if err != nil {
		return []Sword{Green}
	}
	defer file.Close()

	content, err := os.ReadFile(as.swordall)
	if err != nil {
		return []Sword{Green}
	}

	alla := strings.Split(string(content), ",")
	rt := make([]Sword, 0)
	for _, i := range alla {
		if i == "" {
			continue
		}
		ttp, err := strconv.Atoi(i)
		if err != nil || ttp < 0 || ttp > 6 {
			ttp = 0
		}
		rt = append(rt, Sword(ttp))
	}

	rts := make(map[Sword]bool)
	for _, i := range rt {
		rts[i] = true
	}

	rtt := make([]Sword, 0, len(rts))
	for k := range rts {
		rtt = append(rtt, k)
	}
	return rtt
}

func (as *AncientSword) Change(newsword Sword) int {
	tmp := as.ReadAll()
	found := false
	for _, item := range tmp {
		if item == newsword {
			found = true
			break
		}
	}
	if !found {
		return 1
	}

	file, err := os.Create(as.swordnow)
	if err != nil {
		return 1
	}
	defer file.Close()

	_, err = file.WriteString(strconv.Itoa(int(newsword)))
	if err != nil {
		return 1
	}
	return 0
}

func (as *AncientSword) Add(addsword Sword) {
	file, err := os.OpenFile(as.swordall, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return
	}
	defer file.Close()

	_, err = file.WriteString("," + strconv.Itoa(int(addsword)))
	if err != nil {
		return
	}
}

func (s Sword) String() string {
	switch s {
	case Green:
		return "青剑"
	case Red:
		return "赤剑"
	case Purple:
		return "紫剑"
	case Black:
		return "黑剑"
	case White:
		return "白剑"
	case RealRed:
		return "红剑"
	case Blue:
		return "蓝剑"
	default:
		return "紫剑"
	}
}

File: lib/logic/asword.dart
import 'dart:io';

enum Sword {
  green,
  red,
  purple,
  black,
  white,
  realRed,
  blue;

  @override
  String toString() {
    switch (this) {
      case Sword.green:
        return "青剑";
      case Sword.red:
        return "赤剑";
      case Sword.purple:
        return "紫剑";
      case Sword.black:
        return "黑剑";
      case Sword.white:
        return "白剑";
      case Sword.realRed:
        return "红剑";
      case Sword.blue:
        return "蓝剑";
    }
  }
}

class AncientSword {
  final String swordNow;
  final String swordAll;

  AncientSword(this.swordNow, this.swordAll);

  /// 获取当前剑
  Sword getNow() {
    return readNow();
  }

  /// 获取所有剑
  List<Sword> getAll() {
    return readAll();
  }

  /// 读取当前剑
  Sword readNow() {
    // 确保文件存在
    try {
      File(swordNow).openWrite(mode: FileMode.append);
    } catch (e) {
      // 如果无法打开文件，返回默认值
    }

    File file;
    try {
      file = File(swordNow);
      if (!file.existsSync()) {
        file.createSync(recursive: true);
      }
    } catch (e) {
      return Sword.green;
    }

    try {
      String content = file.readAsStringSync();
      if (content.isEmpty) {
        return Sword.green;
      }
      
      int value = int.tryParse(content.trim()) ?? -1;
      if (value >= 0 && value <= Sword.values.length - 1) {
        return Sword.values[value];
      } else {
        return Sword.green;
      }
    } catch (e) {
      return Sword.green;
    }
  }

  /// 读取所有剑
  List<Sword> readAll() {
    // 确保文件存在
    try {
      File(swordAll).openWrite(mode: FileMode.append);
    } catch (e) {
      // 如果无法打开文件，返回默认值
    }

    File file;
    try {
      file = File(swordAll);
      if (!file.existsSync()) {
        file.createSync(recursive: true);
      }
    } catch (e) {
      return [Sword.green];
    }

    try {
      String content = file.readAsStringSync();
      if (content.isEmpty) {
        return [Sword.green];
      }

      List<String> allSwords = content.split(',');
      List<Sword> result = <Sword>[];

      for (String item in allSwords) {
        if (item.isEmpty) continue;
        
        int? value = int.tryParse(item.trim());
        if (value != null && value >= 0 && value <= Sword.values.length - 1) {
          result.add(Sword.values[value]);
        } else {
          // 如果解析失败，默认添加green
          result.add(Sword.green);
        }
      }

      // 去除重复项
      Set<Sword> uniqueSwords = result.toSet();
      return uniqueSwords.toList();
    } catch (e) {
      return [Sword.green];
    }
  }

  /// 更改当前剑
  int change(Sword newSword) {
    List<Sword> swords = readAll();
    bool found = false;

    for (Sword sword in swords) {
      if (sword == newSword) {
        found = true;
        break;
      }
    }

    if (!found) {
      return 1; // 表示失败
    }

    File file;
    try {
      file = File(swordNow);
      file.writeAsStringSync(newSword.index.toString());
    } catch (e) {
      return 1; // 表示失败
    }

    return 0; // 表示成功
  }

  /// 添加新剑
  void add(Sword addSword) {
    File file;
    try {
      file = File(swordAll);
      if (!file.existsSync()) {
        file.createSync(recursive: true);
      }
      
      // 先读取现有内容
      String existingContent = file.readAsStringSync();
      String separator = existingContent.isEmpty ? '' : ',';
      file.writeAsStringSync('$existingContent$separator${addSword.index}');
    } catch (e) {
      // 如果出错不做任何操作
    }
  }
}